lib/MusicBrainz/DataStore/Redis.pm

```diff
@@ -40,39 +40,48 @@ has '_json' => (
     default => sub { return JSON->new->allow_nonref->ascii; }
 );
 
+sub _prepare_key {
+    my ($self, $key) = @_;
+    return encode('utf-8', $self->prefix . $key);
+}
+
+sub _decode_value {
+    my ($self, $value) = @_;
+    return defined $value ? $self->_json->decode($value) : undef;
+}
+
```

Comment:
Gave these repeated code snippets their own methods. They are short but not necessarily trivial and might conceivably change at some point.